beancount/query/query_compile.py

```diff
@@ -90,6 +91,16 @@ def __init__(self, value):
     def __call__(self, _):
         return self.value
 
+class EvalAmount(EvalNode):
+    __slots__ = ('decimal','currency')
+
+    def __init__(self, decimal,currency):
+        super().__init__(type(amount.Amount))
```

Comment:
`type(amount.Amount)` seems wrong here, it should simply be `amount.Amount`.

Reply:
removed in latest.